doc/classes/EditorUndoRedoManager.xml

```diff
@@ -88,6 +88,13 @@
 				The way undo operation are ordered in actions is dictated by [param backward_undo_ops]. When [param backward_undo_ops] is [code]false[/code] undo option are ordered in the same order they were added. Which means the first operation to be added will be the first to be undone.
 			</description>
 		</method>
+		<method name="force_correct_history">
+			<return type="void" />
+			<description>
+				Forces the next operation (e.g. [method add_do_method]) to use the action's history rather than guessing it from the object. This is sometimes needed when a history can't be correctly determined, like for a nested resource that doesn't have a path yet.
+				This method should only be used when absolutely necessary, otherwise it might cause invalid history state. For most of complex cases, the [code]custom_context[/code] parameter of [method create_action] is sufficient.
+			</description>
```

Comment:
The name of the method is a bit confusing IMO, especially with the description that using it may lead to invalid history state. The name sounds like it will force the history to be correct, so why would it be invalid?

I'm not so familiar with API and the concept of multiple histories / contexts, but from the description maybe `set_history_to_pending_action`?

Reply:
`set` implies setting a specific value.
Maybe `force_fixed_history()` or something similar to convey that it's going to use predetermined history ID instead of guessing.

Reply:
`force_history_from_pending_action`? A bit long :D 

I'm fine with `force_fixed_history` if you think that's better. I have no strong opinion, all names seem a bit confusing but it's a hacky API so that's to be expected.